phoenix-core/src/main/java/org/apache/phoenix/compile/WhereCompiler.java

```diff
@@ -154,14 +156,18 @@ public static Expression compile(StatementContext context, FilterableStatement s
         }
         
         if (context.getCurrentTable().getTable().getType() != PTableType.PROJECTED && context.getCurrentTable().getTable().getType() != PTableType.SUBQUERY) {
-            expression = WhereOptimizer.pushKeyExpressionsToScan(context, statement, expression, extractedNodes);
+            Set<HintNode.Hint> hints = null;
+            if(statement.getHint() != null){
+                hints = statement.getHint().getHints();
+            }
+            expression = WhereOptimizer.pushKeyExpressionsToScan(context, hints, expression, extractedNodes, minOffset);
         }
         setScanFilter(context, statement, expression, whereCompiler.disambiguateWithFamily);
 
         return expression;
     }
     
-    private static class WhereExpressionCompiler extends ExpressionCompiler {
+    public static class WhereExpressionCompiler extends ExpressionCompiler {
```

Comment:
Just checking if this intended or accidental..

Reply:
Intended.  As this is now called from OffsetCompiler